### DISCUSSION TITLE
Should update throw NotFoundError

### DISCUSSION THREAD

**Discussion by rjmackay** [sensedeep/dynamodb-onetable#288] [Q&A] [answered]
I notice update throws NotFoundError when working with unique fields, but without unique fields it just throws a ConditionalCheckFailedException, event if exists = true.

Is this expected or should it throw a specific error type for non-existent records?

**Comment by mobsense** [comment, marked as answer]
If exists == true, that means OneTable should do a condition check to ensure the item exists before updating. If not, it throws a ConditionCheck exception. That is expected.

**Reply by ganeshan102** [reply]
@mobsense Hope it's not required any additional cost to check exists( { "ConditionExpression": "(attribute_exists(#_0)) and (attribute_exists(#_1))) already instead of updating directly.

**Reply by mobsense** [reply]
That check is part of DynamoDB and happens at no extra cost.
